### PR TITLE
Return filter_files back

### PR DESCRIPTION
**Pull request by vmarkovtsev**
Signed-off-by: Vadim Markovtsev <vadim@sourced.tech>

### CODE REVIEW

**Comment by vmarkovtsev** [comment]
@zurk Can you please add tests here

File: lookout/core/lib.py [hunk 1]
```diff
@@ -195,3 +218,38 @@ def parse_files(filepaths: Iterable[str], line_length_limit: int, overall_size_l
...
+    :param overall_size_limit: maximum cumulative files size in bytes. \
+                               The files are discarded after reaching this limit.
+    :param random_state: random generator state for shuffling the files.
+    :param log: logger to use to report the number of excluded files.
+    :return: files passed through the filter and the number of files which were excluded.
+    """
+    files_by_path = {f.path: f for f in files}
+
+    def content_getter(key):
+        return files_by_path[key]
```
Comment (vmarkovtsev) [inline lookout/core/lib.py]:
```suggestion
        return files_by_path[key].content
```

**Comment by zurk** [comment]
I also delete parse_files because we do not use it anywhere. If we need in the future it is easy to restore.

**Comment by zurk** [comment]
Also I suggest to create a class for all these `filter_*` fuctions and move parameters to `__init__()` if you are fine with that. It also helps to make work with logging better.

**Comment by vmarkovtsev** [comment]
@zurk Please return parse_files back, it will be needed in the future.

**Comment by vmarkovtsev** [comment]
Yep, having a class here makes sense but let's postpone for now.

**Comment by vmarkovtsev** [comment]
Thank you @zurk for doing the dirty work

**Pull request merged**